src/components/FileUploadModal/index.tsx

```diff
@@ -0,0 +1,117 @@
+import { Button, Form, Tabs } from "antd";
+import { RcFile } from "antd/lib/upload";
+import React, { useState } from "react";
+import { ActionCreator } from "redux";
+
+import {
+    ClearSimFileDataAction,
+    RequestLocalFileAction,
+} from "../../state/trajectory/types";
+import {
+    SetErrorAction,
+    SetViewerStatusAction,
+} from "../../state/viewer/types";
+
+import CustomModal from "../CustomModal";
+import UrlUploadForm from "./url-upload-form";
+import LocalFileUpload from "../LocalFileUpload";
+import uploadFiles from "./upload-local-files";
+
+import styles from "./style.css";
+
+interface FileUploadModalProps {
+    setIsModalVisible: (isModalVisible: boolean) => void;
+    loadLocalFile: ActionCreator<RequestLocalFileAction>;
+    setViewerStatus: ActionCreator<SetViewerStatusAction>;
+    clearSimulariumFile: ActionCreator<ClearSimFileDataAction>;
+    setError: ActionCreator<SetErrorAction>;
+}
+
+const FileUploadModal: React.FC<FileUploadModalProps> = ({
+    setIsModalVisible,
+    clearSimulariumFile,
+    loadLocalFile,
+    setViewerStatus,
+    setError,
+}) => {
+    const [openTab, setOpenTab] = useState("device");
+    const [noUrlInput, setNoUrlInput] = useState(true);
+    const [fileList, setFileList] = useState<RcFile[]>([]);
+    const [urlForm] = Form.useForm();
+
+    const closeModal = () => {
+        setIsModalVisible(false);
+    };
+
+    const onUrlInput = (event: React.ChangeEvent<HTMLInputElement>) => {
+        setNoUrlInput(event.target.value.length === 0);
```

Comment:
so the form itself takes care of the value here? you're just keeping track of wether something has been entered?

Reply:
Right, the only thing we care about at this level is whether or not any input is present, to decide whether or not to disable the Load button. This component could instead be directly responsible for the code that loads from a url (and perhaps should be for consistency, since loading from device is now handled at this level and not in a child), in which case it would want to store the whole string. But there's so little actual code involved that I don't think it's a huge deal either way.